### DISCUSSION TITLE
Random - Lost Connection to Server - with RHEL/Fedora

### DISCUSSION THREAD

**Discussion by aardvarkglory** [OctoFarm/OctoFarm#1350] [Q&A]
When installing on RHEL/Clone or Fedora, I get random disconnects and a message that says "Lost Connection to Server"  I am using NodeJS 18.  I have tried MongoDB 5 and 6 with the same issue.  I do not get this issue in Debian or Ubuntu.  Have also tried different hardware.  Any help would be greatly appreciated.

**Comment by aardvarkglory** [comment]
Here is an example of what it is doing

![octofarm - lost connection](https://user-images.githubusercontent.com/3628650/207741806-494f690c-96b9-408d-a265-8f4af07caf39.jpg)

**Reply by NotExpectedYet** [reply]
Can you provide any logs? Might be better to open a ticket for this.

**Reply by NotExpectedYet** [reply]
I've never used fedora but if ubuntu works that should as well

**Reply by aardvarkglory** [reply]
After playing with this issue for a few days, I found out that it was SELinux that is causing the issue and is preventing PM2 from getting a PID from the OctoFarm home folder.  There are 2 ways to fix this issue.

1. Disable SELinux.  This is fine if security is not an issue.

2. This is the route I went and works by editing the OctoFarm PM2 systemd file.  (You will need to change the username for the PM2 service.)
sudo vim /etc/systemd/system/pm2-username.service

Add the following line:
Environment=PM2_PID_FILE_PATH=/run/pm2.pid

Replace: PIDFile=/username/.pm2/pm2.pid to: PIDFile=/run/pm2.pid


I am not sure how to make SELinux ignore the PM2 service and OctoFarm home folder.  If anyone out there has knowledge on how to do this, please share.  In the mean time, hope this helps those that would like to use OctoFarm with RHEL/Clones or Fedora.

**Reply by NotExpectedYet** [reply]
Cheers for the additional information! 

It might be a limitation with pm2 and SELinux, the pm2 service would have to support it. 

You don't need to use a pm2 service though, you could create a systemd one, but currently the restart button / server only work with pm2. 

I can enable support for a custom restart command if it helps? Seems easier than shoe horning pm2 into this.

**Reply by NotExpectedYet** [reply]
I've added a sample systemd service that should start OctoFarm up on boot.
